### ISSUE TITLE
Misleading error message when ambiguity between a function and module is introduced

### ISSUE THREAD

**Issue by davidbarsky** [rust-lang/rust#71406]
Given this code ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=54b81f2b1faaa57f89be44225cf26963)):

```rust
use tokio::sync::mpsc;

fn main() {
    let (tx, rx) = mpsc::channel::new(1);
}
```

I expected that `rustc` would indicate that `new()` is not function but `channel(1)` _is_.  Instead, I got:

```rust
   Compiling playground v0.0.1 (/playground)
error[E0433]: failed to resolve: could not find `channel` in `mpsc`
 --> src/main.rs:4:26
  |
4 |     let (tx, rx) = mpsc::channel::new(1);
  |                          ^^^^^^^ could not find `channel` in `mpsc`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0433`.
error: could not compile `playground`.

To learn more, run the command again with --verbose.
```

A diagnostic along the lines of `new(..)` isn't a function, but `channel(..)` is would be helpful. 

### Meta

`rustc --version --verbose`:
```
❯ rustc --version --verbose
rustc 1.42.0 (b8cedc004 2020-03-09)
binary: rustc
commit-hash: b8cedc00407a4c56a3bda1ed605c6fc166655447
commit-date: 2020-03-09
host: x86_64-apple-darwin
release: 1.42.0
LLVM version: 9.0
```

</p>
</details>

**Comment by contrun** [comment]
I tried to resolve this issue with  https://github.com/rust-lang/rust/pull/71419

The error message now is
```
`channel` in `mpsc` is a concrete value (function, constant), not a module or Struct you specified
```
Any informative error message suggestions?

**Comment by davidbarsky** [comment]
@contrun Thanks for trying to address this. I'll comment on the PR you opened.
